### ISSUE TITLE
jsonencode() incorrectly dumps containers that are present twice

### ISSUE THREAD

**Issue by ZyX-I** [vim/vim#587]
Consider the following script:

```
vim -u NONE --cmd 'let l=[1, 2]' --cmd 'echo jsonencode([l, l])'
```

This will echo `[[1, 2], []]` which is not correct because `[l, l]` in this case can be dumped correctly and dumped to `[[1, 2], [1, 2]]`.

**Comment by brammool** [comment]
Nikolai Aleksandrovich Pavlov wrote:

> Consider the following script:
> 
> ```
> vim -u NONE --cmd 'let l=[1, 2]' --cmd 'echo jsonencode([l, l])'
> ```
> 
> This will echo `[[1, 2], []]` which is not correct because `[l, l]` in this case can be dumped correctly and dumped to `[[1, 2], [1, 2]]`.

Forgot to reset the copyID.

## 

hundred-and-one symptoms of being an internet addict:
30. Even though you died last week, you've managed to retain OPS on your
    favorite IRC channel.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by ZyX-I** [comment]
Fixed by 7.4.1166.
